### ISSUE TITLE
addItemToOrder does not always merge existing order line

### ISSUE THREAD

**Issue by christopheblin** [vendure-ecommerce/vendure#1612]
**Describe the bug**
Each time I addItemToOrder, it creates a new line instead of merging the existing line

![Capture d’écran 2022-06-07 à 15 32 22](https://user-images.githubusercontent.com/11044877/172396888-70d1370f-3bfa-4fd2-97f8-6b7d1dc70e6d.png)

**To Reproduce**

Create a productVariant with a customField named customPrice with nullable: true, type: 'int', min: 0,

call addItemToOrder multiple times without specifying customPrice, it will create more than one orderLine

**Expected behavior**
Should merge with existing order line

**Environment (please complete the following information):**
 - @vendure/core version: 1.5.1

**Additional context**

For me, the bug is located in `private async customFieldsAreEqual(` of order-modifier.js

Basically, if you do not provide the customField, input["customPrice"] === undefined whereas existingValue === null, so comparing with JSON.stringify will not work

![Capture d’écran 2022-06-07 à 15 53 10](https://user-images.githubusercontent.com/11044877/172397919-68b2f2f8-5b01-44f7-9db7-1d70cf80d2b2.png)

**Comment by christopheblin** [comment]
Sorry, the problem is not with int customField, but with `type: 'boolean', defaultValue: false,`

When you have nullable int, the undefined is correctly matched to null

However, when you have a defaultValue on the field, it is not used (i.e instead of looking at inputCustomFields[key], you must also look at def.defaultValue)

**Comment by christopheblin** [comment]
Maybe something like : 

                const defaultValueMatch = inputCustomFields?.[key] === undefined && def.defaultValue === existingValue
                if (!valuesMatch && !undefinedMatchesNull && ! defaultValueMatch) {
                    return false;
                }
